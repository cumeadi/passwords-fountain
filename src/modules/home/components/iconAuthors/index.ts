export { IconAuthors } from './iconAuthors.component';
