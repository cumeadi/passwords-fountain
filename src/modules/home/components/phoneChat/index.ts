export { PhoneChat } from './phoneChat.component';
