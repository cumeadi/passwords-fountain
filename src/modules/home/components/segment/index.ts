export { Segment } from './segment.component';
