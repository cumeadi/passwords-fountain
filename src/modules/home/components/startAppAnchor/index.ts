export { StartAppAnchor } from './startAppAnchor.component';
