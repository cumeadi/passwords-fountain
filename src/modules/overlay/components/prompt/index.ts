export { Prompt } from './prompt.component';
