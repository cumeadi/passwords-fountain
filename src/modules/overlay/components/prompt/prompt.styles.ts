import { h } from 'preact';
import { styled, setPragma } from 'goober';
import { theme, typography } from '@/common/theme';
setPragma(h);

export const Wrapper = styled('section')`
    position: fixed;
    z-index: ${theme.zIndexes.m};
    top: 76px;
    left: 50%;
    transform: translateX(-50%);
    display: grid;
    width: 100%;
    min-width: 320px;
    max-width: 640px;
    justify-items: center;
    grid-auto-flow: row;
    padding: ${theme.spacing.xl30} ${theme.spacing.m18};
    border: 1px solid ${theme.colors.primaryBlue};
    border-top: 4px solid ${theme.colors.primaryBlue};
    background: ${theme.colors.white};
`;

export const ContentWrapper = styled('div')`
    ${typography.text18}
    display: grid;
    max-width: 480px;
    width: 100%;
    padding: 0 ${theme.spacing.xs6} ${theme.spacing.xs6};
    margin-bottom: ${theme.spacing.m18};
    text-align: center;
`;

export const ControlsWrapper = styled('div')`
    display: grid;
    grid-gap: ${theme.spacing.xl30};
    grid-auto-flow: column;
`;
