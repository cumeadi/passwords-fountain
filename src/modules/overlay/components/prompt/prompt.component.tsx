import { h, VNode } from 'preact';
import { TypedComponent } from '@/common/typings/prop-types';
import PropTypes from 'prop-types';
import { Wrapper, ContentWrapper, ControlsWrapper } from './prompt.styles';

export const Prompt: TypedComponent<Props> = ({
    renderContent,
    renderControls,
}: Props) => {
    return (
        <Wrapper onClick={(e: Event) => e.stopPropagation()}>
            <ContentWrapper>{renderContent()}</ContentWrapper>
            <ControlsWrapper>{renderControls()}</ControlsWrapper>
        </Wrapper>
    );
};

interface Props {
    renderContent: () => VNode | string;
    renderControls: () => VNode;
}

Prompt.propTypes = {
    renderContent: PropTypes.func.isRequired,
    renderControls: PropTypes.func.isRequired,
};
