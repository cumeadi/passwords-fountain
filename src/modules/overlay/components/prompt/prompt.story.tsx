import { h, VNode, Fragment } from 'preact';
import { Prompt } from './prompt.component';
import { withA11y } from '@storybook/addon-a11y';
import { Button } from '@/common/components/button';

export default {
    title: 'Prompt',
    decorators: [withA11y],
};

export const defaultView = (): VNode => (
    <Prompt
        renderMessage={(): string =>
            'Are you sure you want to delete this file?'
        }
        renderControls={(): VNode => (
            <Fragment>
                <Button
                    onClick={(): void => {
                        // empty
                    }}
                >
                    Cancel
                </Button>
                <Button
                    onClick={(): void => {
                        // empty
                    }}
                >
                    Ok
                </Button>
            </Fragment>
        )}
    />
);
