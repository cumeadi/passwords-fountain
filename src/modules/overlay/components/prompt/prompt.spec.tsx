import '@testing-library/jest-dom/extend-expect';
import { Fragment, h, VNode } from 'preact';
import { render } from '@testing-library/preact';
import { Prompt } from './prompt.component';
import { Button } from '../../../../common/components/button';

describe('Prompt', () => {
    it('should render correctly', () => {
        const { asFragment } = render(
            <Prompt
                renderContent={(): string =>
                    'Are you sure you want to delete this file?'
                }
                renderControls={(): VNode => (
                    <Fragment>
                        <Button
                            onClick={(): void => {
                                // empty
                            }}
                        >
                            Cancel
                        </Button>
                        <Button
                            onClick={(): void => {
                                // empty
                            }}
                        >
                            Ok
                        </Button>
                    </Fragment>
                )}
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
