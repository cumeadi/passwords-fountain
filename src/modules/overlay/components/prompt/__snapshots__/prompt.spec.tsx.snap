// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prompt should render correctly 1`] = `
<DocumentFragment>
  <section
    class="go939053875"
  >
    <div
      class="go579113919"
    >
      Are you sure you want to delete this file?
    </div>
    <div
      class="go444860531"
    >
      <button
        class="go3135434547"
        type="button"
      >
        <span
          class="go4062099715"
        >
          Cancel
        </span>
      </button>
      <button
        class="go3135434547"
        type="button"
      >
        <span
          class="go4062099715"
        >
          Ok
        </span>
      </button>
    </div>
  </section>
</DocumentFragment>
`;
